packages/c/sshnpd/src/main.c

```diff
@@ -338,351 +342,33 @@ int main(int argc, char **argv) {
-          atlogger_log(LOGGER_TAG, ATLOGGER_LOGGING_LEVEL_DEBUG, "Executing handle_ping\n");
-          handle_ping(&params, &message, ping_response, &worker, &atclient_lock);
-          break;
-        case NK_SSH_REQUEST:
-          atlogger_log(LOGGER_TAG, ATLOGGER_LOGGING_LEVEL_DEBUG, "Executing handle_ssh_request\n");
-          // permitopen happens first for ssh so we can avoid a bunch of unnecessary tasks
-          permitopen.requested_host = "localhost";
-          permitopen.requested_port = params.local_sshd_port;
-          if (!should_permitopen(&permitopen)) {
-            atlogger_log(LOGGER_TAG, ATLOGGER_LOGGING_LEVEL_DEBUG, "Ignoring request to localhost:%d\n",
-                         params.local_sshd_port);
-            // TODO notify daemon doesn't permit connections to $requested_host:$requested_port
-            break;
-          }
-          handle_ssh_request(&worker, &atclient_lock, &params, &is_child_process, &message, signingkey);
-          if (is_child_process) {
-            atlogger_log(LOGGER_TAG, ATLOGGER_LOGGING_LEVEL_DEBUG, "Exiting child process\n");
-            atclient_monitor_message_free(&message);
-            return;
-          }
-          break;
-        case NK_NPT_REQUEST:
-          atlogger_log(LOGGER_TAG, ATLOGGER_LOGGING_LEVEL_DEBUG, "Executing handle_npt_request\n");
-          // No permitopen here... since we need to parse the json first in order to check, it happens inside
-          // handle_npt_request
-          handle_npt_request(&worker, &atclient_lock, &params, &is_child_process, &message, signingkey);
-          break;
-        case NK_GRACEFUL_SHUTDOWN:
-#ifdef SSHNPD_ENABLE_TESTING_SHUTDOWN_NOTIFICATION
-#warning BINARY COMPILED WITH SHUTDOWN NOTIFICATION ENABLED NOT FOR PRODUCTION USE
-          atlogger_log(LOGGER_TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "TRIGGERING GRACEFUL SHUTDOWN\n");
-          should_run = 0;
-#endif
-          break;
-        case NK_NONE:
-          break;
-        }
-      } else {
-        atlogger_log(LOGGER_TAG, ATLOGGER_LOGGING_LEVEL_DEBUG, "Skipping notification (no decryptedvalue): %s\n",
-                     message.notification->id);
-      }
-      break;
-    } // end of case ATCLIENT_MONITOR_MESSAGE_TYPE_NOTIFICATION
-    } // end of switch
-    atclient_monitor_message_free(&message);
-  } // end of while loop
-}
-
-static int lock_atclient(void) {
-  int ret = pthread_mutex_lock(&atclient_lock);
-  if (ret != 0) {
-    atlogger_log(LOGGER_TAG, ATLOGGER_LOGGING_LEVEL_ERROR,
-                 "Failed to get a lock on atclient for sending a notification\n");
-  } else {
-    atlogger_log(LOGGER_TAG, ATLOGGER_LOGGING_LEVEL_DEBUG, "Locked the atclient\n");
-  }
-  return ret;
-}
-
-static int unlock_atclient(int ret) {
-  ret = pthread_mutex_unlock(&atclient_lock);
-  if (ret != 0) {
-    atlogger_log(LOGGER_TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "Failed to release atclient lock\n");
-    exit(1);
+    atcommons_memlist_success_free(&memlist);
```

Comment:
we are using success here to handle freeing the child process memory, since we can optionally not free certain things (like the mbedtls sockets which will get closed if we do) on a success free.

Reply:
Doing this is mostly pointless, but if we need to memcheck srv, there will be unavoidable leaks on that process, but at least we can minimize the number of unavoidable ones, which may help find leaks that actually matter.